client/sagas/err-saga.js

```diff
@@ -1,11 +1,27 @@
+import combineSagas from '../../common/utils/combine-sagas';
+import { ofType } from '../../common/utils/get-actions-of-type';
 import { makeToast } from '../../common/app/toasts/redux/actions';
+import types from '../../common/app/redux/types';
 
-export default function errorSaga(action$) {
+function errorSaga(action$) {
   return action$
     .filter(({ error }) => !!error)
+    .filter(({ type }) => type !== types.babelError)
     .map(({ error }) => error)
     .doOnNext(error => console.error(error))
     .map(() => makeToast({
       message: 'Something went wrong, please try again later'
     }));
 }
+
+/* this is for catching transpilation errors for the React challenges */
+function babelErrorSaga(action$) {
```

Comment:
These should be caught right at the compiler, this file is a generic error handler